src/viam/sdk/tests/test_resource.cpp

```diff
@@ -0,0 +1,243 @@
+#define BOOST_TEST_MODULE test module test_resource
+#include <boost/test/included/unit_test.hpp>
+
+#include <google/protobuf/struct.pb.h>
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/common/proto_type.hpp>
+#include <viam/sdk/config/resource.hpp>
+#include <viam/sdk/referenceframe/frame.hpp>
+#include <viam/sdk/resource/resource_api.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+#include <viam/sdk/spatialmath/orientation.hpp>
+#include <viam/sdk/spatialmath/orientation_types.hpp>
+
+namespace viam {
+namespace sdktests {
+
+using namespace viam::sdk;
+using google::protobuf::Struct;
+using google::protobuf::Value;
+
+BOOST_AUTO_TEST_CASE(test_api) {
+    API api1;
+    api1.set_namespace("ns");
+    BOOST_CHECK_EQUAL(api1.type_namespace(), "ns");
+    api1.set_resource_type("component");
+    BOOST_CHECK_EQUAL(api1.resource_type(), "component");
+    api1.set_resource_subtype("st");
+    BOOST_CHECK_EQUAL(api1.resource_subtype(), "st");
+    BOOST_CHECK_EQUAL(api1.to_string(), "ns:component:st");
+    BOOST_CHECK(!api1.is_service_type());
+    BOOST_CHECK(api1.is_component_type());
+
+    API api2 = API::from_string("ns:service:st");
+    BOOST_CHECK_EQUAL(api1.type_namespace(), "ns");
+    BOOST_CHECK_EQUAL(api1.resource_subtype(), "st");
```

Comment:
(nit) can we move the `api1` tests here above with the `api1` block?

Reply:
Oh oops! These are supposed to be for `api2`! Thanks for catching that